### PR TITLE
#453-implement-connect-social-media-buttons

### LINKED ISSUE #453: Implement “Connect Social Media” Buttons for Users and Organization
**Is your feature request related to a problem? Please describe.**
- Social Media Connections haven't been implemented yet for organizations. 
- Implement "Connect Social Media" actions in both Users and Org 
**(Don't implement the organization settings social media buttons for now)**

- [ ] Implement Connect Social Media for Users Settings

**Describe the solution you'd like**
reference : https://firebase.google.com/docs/auth/web/account-linking

image
image

### PR DESCRIPTION
**Pull request by ozair-dev**
## Description
Here is the minimal implementation of linking auth providers with existing accouns. Please have a look and let me know if it needs some more work

## Related Issue

Fixes #453 

## How Has This Been Tested?
Tested on my machine locally with firebase emulator

## Screenshots or GIF (In case of UI changes):
Screenshot from 2022-10-19 22-12-48

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ozair-dev** [comment]
U there?

**Review by Sarfraz-droid — DISMISSED** [review]

File: src/components/Profile/ConnectSocials/index.js [hunk 1]
```diff
@@ -9,66 +9,100 @@ import TwitterIcon from "@material-ui/icons/Twitter";
...
-import { useDispatch } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { useFirebase } from "react-redux-firebase";
 
+const SocialButton = ({ Icon, isLinked, ...props }) => {
+  const classes = useStyles();
+  return (
+    <Box
+      {...props}
+      className={`${classes.link} ${isLinked ? classes.linkDisabled : ""}`}
```
Comment (Sarfraz-droid) [inline src/components/Profile/ConnectSocials/index.js:20]: use `isLinked && classes.linkDisabled` insted

**Comment by ozair-dev** [comment]
done, but not tested

**Review by Sarfraz-droid — APPROVED** [review]

**Review by shahaab — APPROVED** [review]

**Comment by Sarfraz-droid** [comment]
@ozair-dev can you please test the providers? if possible?

**Comment by ozair-dev** [comment]
The providers are already tested using firebase emulaor suite.

But I encourage you to test them yourself to get the better picture of how they're implemented. We could make some changes, like: 

- show error when linking a provider that is already linked
- show error when linking social media account that is already linked with another account.
- Allow user to unlink a provider etc

**Comment by Sarfraz-droid** [comment]
@ozair-dev we can do the above changes in next contributions.

For this pr, The links are working but if we connect multiple providers then there comes an **Auth Internal Emulator Error**.. As far as I think, its has something to do with the emulator itself. One way to fix this issue is to just reload the emulator after connection

**Review by Sarfraz-droid — APPROVED** [review]

**Comment by Sarfraz-droid** [comment]
WTrUiIzI5I

**Review by Maahi10001 — APPROVED** [review]

**Pull request merged**